package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public abstract class BasePage {
    static WebDriver driver;

    public void setDriver(WebDriver wd) {
        driver = wd;
    }

    @FindBy(xpath = "//div[contains(@class,'error')]")
    List<WebElement> listErrors;

    public boolean isTextInElementPresent(WebElement element, String text) {
        try {
            return new WebDriverWait(driver, Duration.ofSeconds(5))
                    .until(ExpectedConditions
                            .textToBePresentInElement(element, text));
        } catch (RuntimeException e) {
            e.printStackTrace();
            System.out.println("created exeption");
        }
        return false;
    }

    public boolean isTextInErrorPresent(String text) {
        if (listErrors == null || listErrors.isEmpty())
            return false;
        for (WebElement element : listErrors) {
            if (element.getText().contains(text))
                return true;
        }
        return true;
    }

    public boolean isElementDisplayed(WebElement element) {
        return element.isDisplayed();
    }

    public void pause(int time) {
        try {
            Thread.sleep(time);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }
}
